jekyll/_posts/guides/2018-09-21-usage-of-the-matrix-js-sdk.md

```diff
@@ -0,0 +1,199 @@
+    console.log(response.access_token);
+});
+
+```
+
+In any case, once logged in either with a password or an access token, the client can get the current access token via:
+
+```javascript
+console.log(client.getAccessToken());
+```
+
+## Sync and Listen
+
+Let's perform a first sync, and listen for the response:
+
+```javascript
+client.once('sync', function(state, prevState, res) {
+    console.log(state); // state will be 'PREPARED' when the client is ready to use
+});
+```
+
+Once the sync is complete, we can add listeners for events. To listen to ALL incoming events:
+
+```javascript
+client.on("event", function(event){
+    console.log(event.getType());
+    console.log(event);
+})
+```
+
+But that would be a lot of traffic, and too much for our simple example. Let's just listen to events happening on the timeline of rooms for which our user is a member:
+
+```javascript
+client.on("Room.timeline", function(event, room, toStartOfTimeline) {
+    console.log(event.event);
+});
+```
+
+## Access the Store
+
+When we created a new client with `sdk.createClient()`, an instance of the default store, `MatrixInMemoryStore` was created and enabled.
+
+To access the store, we use accessor methods. For example, to get a list of rooms in which our user is joined:
+
+```javascript
+// client.client.getRooms() returns an array of room objects
+var rooms = client.getRooms();
+rooms.forEach(room => {
+    console.log(room.roomId);
+});
+```
+
+More usefully, we could get a list of members for each of these rooms:
+
+```javascript
+var rooms = client.getRooms();
+rooms.forEach(room => {
+    var members = room.getJoinedMembers();
+    members.forEach(member => {
+        console.log(member.name);
+    });
+});
+```
+
+For each room, we can inspect the timeline in the store:
+
+```javascript
+var rooms = client.getRooms();
+rooms.forEach(room => {
+    room.timeline.forEach(t => {
+        console.log(JSON.stringify(t.event.content));
+    });
+});
+```
+
+## Send messages to rooms
+
+To send a message, we create a content object, and specify a room to send to:
+
+```javascript
+var content = {
+    "body": "Hello World",
+    "msgtype": "m.text"
+};
+
+client.sendEvent("!jhpZBTbckszblMYjMK:matrix.org", "m.room.message", content, "").then((res) => {
+   // message sent successfully
+}).catch((err) => {
+    console.log(err);
+}
+```
+
+Knowing this, we can build a bot which just echos back any message starting with a "!"
+
+```javascript
+var testRoomId = "!jhpZBTbckszblMYjMK:matrix.org";
+
+client.on("Room.timeline", function(event, room, toStartOfTimeline) {
+    // we know we only want to respond to messages
+    if (event.getType() !== "m.room.message") {
+        return;
+    }
+
+    // we are only intested in messages from the test room, which start with "!"
+    if (event.getRoomId() === testRoomId && event.getContent().body[0] === '!') {
+        sendNotice(event.event.content.body);
+    }
+});
+
+function sendNotice(body) {
```

Comment:
might want to comment what a notice is and why we're sending it here (i.e. 'cos we're a bot)